lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -357,16 +353,15 @@ ValueDecl *DerivedConformance::deriveRawRepresentable(TypeChecker &tc,
                                                       Decl *parentDecl,
                                                       NominalTypeDecl *type,
                                                       ValueDecl *requirement) {
-  // Check preconditions. These should already have been diagnosed by
-  // type-checking but we may still get here after recovery.
+  // Check preconditions for RawRepresentable conformance.
```

Comment:
It would be great to put the precondition check in its own function, since the same logic is duplicated in the two overloads of DerivedConformance::deriveRawRepresentable()